File: src/mobile/src/models/supplier.ts
export interface Supplier {
  id: string;
  supplierName: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
  stateDivision?: string;
  city?: string;
  country?: string;
  logoUrl?: string;
  isActive: boolean;
  rowVersion?: string;
  createdAtUtc?: string;
  updatedAtUtc?: string;
}

export interface SupplierListPayload {
  items?: Supplier[];
  page?: number;
  pageSize?: number;
  totalCount?: number;
  totalPages?: number;
}

File: src/api/Entities/Supplier.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FarmPlus.Api.Entities;

[Table("Suppliers")]
public class SupplierEntity : EntityBase<Guid>
{
    [Required]
    [MaxLength(50)]
    public required string SupplierName { get; set; }

    [EmailAddress]
    [MaxLength(50)]
    public string? Email { get; set; }

    [MaxLength(100)]
    public string? PhoneNumber { get; set; }

    [MaxLength(100)]
    public string? Address { get; set; }

    [MaxLength(100)]
    public string? StateDivision { get; set; }

    [MaxLength(50)]
    public string? City { get; set; }

    [MaxLength(50)]
    public string? Country { get; set; }

    [MaxLength(100)]
    public string? LogoUrl { get; set; }

    [Required]
    public bool IsRequired { get; set; }

}
